pywr-core/src/parameters/profiles/weekly.rs

```diff
@@ -0,0 +1,459 @@
+use crate::model::Model;
+use crate::parameters::{Parameter, ParameterMeta};
+use crate::scenario::ScenarioIndex;
+use crate::state::State;
+use crate::timestep::Timestep;
+use crate::PywrError;
+use std::any::Any;
+use time::{util::is_leap_year, Date};
+
+pub enum WeeklyInterpDay {
+    First,
+    Last,
+}
+
+// A weekly profile can be 52 or 53 week long
+enum WeeklyProfileSize {
+    FiftyTwo([f64; 52]),
+    FiftyThree([f64; 53]),
+}
+
+impl WeeklyProfileSize {
+    /// Get the week position in a calendar year from date. The position starts from 0 on the
+    /// first week day and ends with 1 on the last week day.
+    fn current_pos(&self, date: &Date) -> f64 {
```

Comment:
I initially implemented the parameter using the ISO format but scrapped it because it was different from pywr 1. I guess wWe could add a flag called  `use_iso_date` to calculate the week number in a different way.